python/pyspark/sql/connect/session.py

```diff
@@ -840,7 +848,13 @@ def stop(self) -> None:
                 # meaning that you can stop local mode, and restart the Spark Connect
                 # client with a different remote address.
                 if PySparkSession._activeSession is not None:
-                    PySparkSession._activeSession.stop()
+                    try:
+                        PySparkSession._activeSession.stop()
+                    except Exception as e:
+                        warnings.warn(
+                            "session.stop(): Local Spark Connect Server could not be stopped. "
+                            f"Error: ${e}"
+                        )
```

Comment:
I'm wondering if this one actually needs to be wrapped because. If I'm reading this correctly, the PySparkSession._activeSession is actually a Java Session.

![image](https://github.com/apache/spark/assets/3421/4df21480-6b6a-48ab-8708-3c6e960883b5)

Reply:
It is the underlying Java session. 

Unclear. Felt safer such as in the case that the JVM is killed out of band. I don't know all the failure modes in this code path.